include/xxhash.h

```diff
@@ -606,41 +609,6 @@ XXH_PUBLIC_API void XXH32_canonicalFromHash(XXH32_canonical_t* dst, XXH32_hash_t
 XXH_PUBLIC_API XXH32_hash_t XXH32_hashFromCanonical(const XXH32_canonical_t* src);
 
 
-#ifdef __has_attribute
-# define XXH_HAS_ATTRIBUTE(x) __has_attribute(x)
-#else
-# define XXH_HAS_ATTRIBUTE(x) 0
-#endif
-
-/* C-language Attributes are added in C23. */
-#if defined(__STDC_VERSION__) && (__STDC_VERSION__ > 201710L) && defined(__has_c_attribute)
-# define XXH_HAS_C_ATTRIBUTE(x) __has_c_attribute(x)
-#else
-# define XXH_HAS_C_ATTRIBUTE(x) 0
-#endif
-
-#if defined(__cplusplus) && defined(__has_cpp_attribute)
-# define XXH_HAS_CPP_ATTRIBUTE(x) __has_cpp_attribute(x)
-#else
-# define XXH_HAS_CPP_ATTRIBUTE(x) 0
-#endif
-
-/*
-Define XXH_FALLTHROUGH macro for annotating switch case with the 'fallthrough' attribute
-introduced in CPP17 and C23.
-CPP17 : https://en.cppreference.com/w/cpp/language/attributes/fallthrough
-C23   : https://en.cppreference.com/w/c/language/attributes/fallthrough
-*/
-#if XXH_HAS_C_ATTRIBUTE(x)
-# define XXH_FALLTHROUGH [[fallthrough]]
-#elif XXH_HAS_CPP_ATTRIBUTE(x)
-# define XXH_FALLTHROUGH [[fallthrough]]
-#elif XXH_HAS_ATTRIBUTE(__fallthrough__)
-# define XXH_FALLTHROUGH __attribute__ ((fallthrough))
-#else
-# define XXH_FALLTHROUGH
-#endif
-
```

Comment:
I found this code has derived from https://github.com/Cyan4973/xxHash.
Keeping differences from the upstream code smaller may be good idea.
CAUTION: I'm just a contributor of this project. So @karelzak, the maintainer can have different comment.

```
#define XXH_FALLTHROUGH FALLTHROUGH 
```

With adding this definition here, we don't have to replace the line using `XXH_FALLTHROUGH;` with `FALLTHROUGH;`.

So the pull request may has three commits:
1. back porting the fix of `#if C_HAS_C_ATTRIBUTE(x)`,
2. moving `XXH_FALLTHROUGH` to `c.h` with renaming to `FALLTHROUGH` (or `UL_FALLTHROUGH`) and adding `#define XXH_FALLTHROUGH FALLTHROUGH` to xxhash.h, and
3. replacing `/* fallthrough */` with FALLTHROUGH

Reply:
LGTM. @karelzak  What do you think?

Reply:
I do not understand the second step. Why must we keep any `XXH_*` macro in the generic `c.h` if the macro is used only in xxhash.h? I think this step is unnecessary.

Reply:
Sorry. Let me withdraw my "LGTM".
we should remove `XXH_` prefix of macros in c.h, or
we should use `UL_` prefix instead of `XXH_` .

Reply:
Ah.. I read the patches in the test branch, but you're clear about the rename in your plan :-) Yes, after renaming it, it makes sense, and I look forward to PR with it ;-)

Reply:
what about `FALLTHROUGH`? Keep it literal or `UL_FALLTHROUGH`?

Reply:
I cannot imagine that any library will define something as generic as FALLTHROUGH, so using it without the UL_ prefix sounds safe. But maybe I'm too optimistic. :-) We will see. Let's start without UL_.